### PR TITLE
DB2

### PR DESCRIPTION
**Pull request by mixmix**
:warning: BREAKING CHANGE

updates:
- :warning: make db2 the default database
- run all tests for both databases
- improved logging of named instances
- upstream fixes to db2 (PR merge required)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,21 +25,21 @@
...
-    "secret-stack": "^6.4.0",
-    "ssb-conn": "^6.0.3",
+    "pull-stream": "^3.7.0",
+    "rimraf": "^3.0.2",
+    "secret-stack": "^6.4.1",
+    "ssb-conn": "^6.0.4",
     "ssb-db": "^20.4.1",
-    "ssb-db2": "^6.2.7",
-    "ssb-keys": "^8.4.1"
+    "ssb-db2": "github:ssbc/ssb-db2#historyStream_live",
```
Comment (mixmix) [inline package.json:33]: TODO
Is a minor release (or patch if you consider missing API a bug?)

Comment (mixmix) [reply package.json:33]: https://github.com/ssbc/ssb-db2/pull/423

File: test/connect.js [hunk 2]
```diff
@@ -24,12 +33,25 @@ test('connect', t => {
...
   })
 
   connect([alice, bob, cherese], { name, friends: true }, (err) => {
-    t.error(err, 'alice an bob connected!')
+    t.error(err, 'happily connected')
+
+    // Test if auto-replication because of friends (follows) + ssb-replicate
+    // is working
+    if (isDB2) return t.end()
+    // currently doesn't work with db2
```
Comment (Powersource) [inline test/connect.js:54]: will this work after the pr is merged?

**Review by Powersource — APPROVED** [review]

File: test/connect.js (hunk 2 above)
Comment (mixmix) [reply test/connect.js:54]: No. there's lower level problems - basically friends fundamentally changed how it worked. This lower version orchestrated replication with friends on connections... that doesn't work with new db2 for some reasons. Maybe friends as a module is broken as it was flume based at this version. Dunno. Anyway it does what it says "connects" but the rest of the test doesn't

**Pull request merged**